### PR TITLE
Optional config file support

### PR DESCRIPTION
**Pull request by snoyberg**

### CODE REVIEW

File: packages/cosmos/src/clap.rs [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+
 use crate::{error::BuilderError, AddressHrp, Cosmos, CosmosBuilder, CosmosNetwork};
 
 /// Command line options for connecting to a Cosmos network
 #[derive(clap::Parser, Clone, Debug)]
 pub struct CosmosOpt {
     /// Which blockchain to connect to for grabbing blocks
     #[clap(long, env = "COSMOS_NETWORK", global = true)]
-    pub network: Option<CosmosNetwork>,
+    pub network: Option<String>,
```
Comment (psibi) [inline packages/cosmos/src/clap.rs:12]: Any reason for moving it to `String` from `CosmosNetwork` ?

File: packages/cosmos/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+/// Configuration overrides for individual network
+pub struct CosmosConfig {
+    path: PathBuf,
+    inner: CosmosConfigInner,
+}
+
+#[derive(serde::Deserialize, Debug)]
+struct CosmosConfigInner {
+    #[serde(default)]
+    network: HashMap<String, NetworkConfig>,
```
Comment (psibi) [inline packages/cosmos/src/config.rs:23]: Same question, I think something like `HashMap<CosmosNetwork, NetworkConfig>` ? Is it to allow a custom network that's not yet supported ?

**Review by psibi — APPROVED** [review]

File: packages/cosmos/src/config.rs (hunk 2 above)
Comment (snoyberg) [reply packages/cosmos/src/config.rs:23]: Hah, exactly, I hadn't finished typing a response to your previous comment. That's the whole point of the switch to `String`.

**Comment by snoyberg** [comment]
And following up with another PR after this to provide some CLI niceties.

**Pull request merged**